containers/ecr-viewer/src/app/services/socialHistoryService.tsx

```diff
@@ -43,29 +49,49 @@ export const evaluateTravelHistoryTable = (
     },
   ];
 
-  const tables = travelHistory
-    .map((act) => {
-      return columnInfo.reduce(
-        (row, { columnName, infoPath, applyToValue }) => {
-          let val = evaluateValue(act, mappings[infoPath]);
-          if (applyToValue) {
-            val = applyToValue(val) ?? "";
-          }
-          return { ...row, [columnName]: { value: val || noData } };
-        },
-        {} as HtmlTableJsonRow[],
-      );
-    })
-    .filter((row) =>
-      Object.values(row).some((v) => (v.value as any) !== noData),
-    );
+  const tables = createTravelHistoryTables(travelHistory, columns, mappings);
 
-  if (!tables.flat().length) return undefined;
+  if (!tables.length) return undefined;
 
   return (
     <JsonTable
-      jsonTableData={{ tables, resultName: "Travel History" }}
+      jsonTableData={{ tables: [tables], resultName: "Travel History" }}
       className="caption-data-title margin-y-0"
     />
   );
 };
+
+const createTravelHistoryTables = (
+  history: Observation[],
+  columns: TravelHistoryColumn[],
+  mappings: PathMappings,
+) => {
+  const tables = history
+    .map((activity) => {
+      const row: HtmlTableJsonRow = {};
+
+      // Populate the row by iterating over the columns
+      columns.forEach(({ columnName, infoPath, applyToValue }) => {
+        let value = evaluateValue(activity, mappings[infoPath]);
+
+        // Apply transformation if needed
+        if (applyToValue) {
+          value = applyToValue(value) ?? "";
+        }
+
+        // Assign the value to the row
+        row[columnName] = { value: value || noData };
+      });
+
+      return row;
+    })
+    .filter((row) => {
+      // Filter out rows with only noData values
+      return Object.values(row).some((column) => {
+        const columnValue = column.value;
+        return columnValue !== noData;
+      });
```

Comment:
```suggestion
      return Object.values(row).some(({value}) => value !== noData);
```
⛳ (feel free to ignore if you don't like)